episodes/3-documentation.md

```diff
@@ -216,49 +215,148 @@ def fahrenheit_to_celsius(temp_f: float) -> float:
 The second set uses **docstrings or similar concepts**. Please compare the two
 (above and below):
 
-```py
+```python
 def fahrenheit_to_celsius(temp_f: float) -> float:
     """
     Converts a temperature in Fahrenheit to Celsius.
 
-    Parameters
-    ----------
-    temp_f : float
-        The temperature in Fahrenheit.
+    Args:
+        temp_f (float): The temperature in Fahrenheit.
 
-    Returns
-    -------
-    float
-        The temperature in Celsius.
+    Returns:
+        float: The temperature in Celsius.
     """
 
     temp_c = (temp_f - 32.0) * (5.0/9.0)
     return temp_c
 ```
 
+Unlike regular comments, docstrings are stored as an attribute (`__doc__`) of the object they document and can be extracted programmatically. 
+
 Docstrings are more powerful than comments:
 
-- Docstrings are automatically extracted when calling the help function.
-- Tools can generate documentation pages automatically from docstrings known as API documentation.
+- Docstrings are automatically extracted when calling the `help` function.
+- Tools (e.g. `mkdocs`, `sphinx`) can generate documentation pages automatically from docstrings known as API documentation.
 
-It is common to write docstrings for functions, classes, and modules.
+A good docstring should describe:
 
-TODO: Introduce how to write docstrings and show an example of API documentation.
-Good docstrings describe:
 - What the function does
 - What goes in (including the type of the input variables)
 - What goes out (including the return type)
 
+There are different docstrings styles, such as Google style, numpy style and reStructered style, with Google style being the most common.
+
+::::::::::::::::::::::::::::::::::::: challenge 
+
+## Exercise: Write a docstring 
```

Comment:
For people who never wrote a docstring you need an example of how you write one. And like I mentioned, I am not a big fan of this exercise, because it is more about learning the syntax, and not about showing the great idea of writing docstrings. But I can live with it :)

Reply:
This exercise is actually to make them discover that type hinting is important. Plus there is an example of docstrings above.

Reply:
removed now!